### ISSUE TITLE
Azure Front Door doesn't support Web Sockets

### ISSUE THREAD

**Issue by mikepfeiffer** [MicrosoftDocs/architecture-center#1891]
Hello team,

Can we call out in this load balancing decision tree (or somewhere else in this doc) that Azure Front Door does not currently support web sockets?

AFD doesn't support Web Sockets:
https://github.com/MicrosoftDocs/azure-docs/issues/29971#issuecomment-486935231

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 936c85d9-c953-2def-c399-77cbda32b24d
* Version Independent ID: 9173c834-2c40-9169-b13b-6273759a2c8a
* Content: [Overview of Azure load-balancing options - Azure Application Architecture Guide](https://docs.microsoft.com/en-us/azure/architecture/guide/technology-choices/load-balancing-overview#feedback)
* Content Source: [docs/guide/technology-choices/load-balancing-overview.md](https://github.com/microsoftdocs/architecture-center/blob/master/docs/guide/technology-choices/load-balancing-overview.md)
* Service: **architecture-center**
* Sub-service: **reference-architecture**
* GitHub Login: @sharad4u
* Microsoft Alias: **pnp**

**Comment by DixitArora-MSFT** [comment]
Hi @mikepfeiffer Thanks for your feedback. We will review and update as appropriate.

**Comment by absheik** [comment]
AB#160195 - Thanks for reporting - this issue is under review

**Comment by HockeyJustin** [comment]
Having just tried front door and found it [doesn't work with SignalR](https://github.com/MicrosoftDocs/azure-docs/issues/29971) service, I'd like to +1 this. SignalR is a critical component for us (and many others), so we can't use front door at the moment.

**Comment by provanguard** [comment]
Apart from SignalR, please list other supported protocols and technologies in the decision tree. I am particularly interested in WCF. What choices do I have when some of the clients use WCF to communicate with some of the micro services behind the gateway/load balancer?

**Comment by crgarcia12** [comment]
https://docs.microsoft.com/en-us/azure/frontdoor/front-door-faq#when-should-we-deploy-an-application-gateway-behind-front-door

<i>
What protocols does Azure Front Door Service support?
Azure Front Door Service supports HTTP, HTTPS and HTTP/2.
</i>

**Comment by absheik** [comment]
https://docs.microsoft.com/en-us/azure/application-gateway/application-gateway-websocket
App Gateway which is a regional service supports WebSockets.  Closing the issue per above pr.

**Comment by dhananjaya-senanayake** [comment]
@absheik , any ETA on having Web Socket Support for Azure Front Door ? https://feedback.azure.com/forums/217313-networking/suggestions/37346371-support-websocket-connections-on-azure-front-door

**Comment by p-bojkowski** [comment]
NO ETA?

**Comment by Ulf-Ason** [comment]
@absheik Why is this issue closed? Does this mean that we should avoid Front Door if we have any plans to use Web sockets and instead use Application Gateway?

**Comment by dfendit** [comment]
Please re-open since it's not fixed yet. What if you need to use Front Door in combination with Application Gateway to let the user utilize Websockets and you have an environment which is spread over multiple regions. Merging CDN into Front Door was a big deal, but when will Websockets be?

**Comment by ShadowXVII** [comment]
They've nuked the feedback.azure.com site. What a pain.

Got a support case open and they said _maybe_ this half of the calendar year. I'll try remember to post here if I hear anything.

**Comment by joshjohanning** [comment]
> They've nuked the feedback.azure.com site. What a pain.
> 
> Got a support case open and they said _maybe_ this half of the calendar year. I'll try remember to post here if I hear anything.

Found this archive.org cache: https://web.archive.org/web/20201021111026/https://feedback.azure.com/forums/217313-networking/suggestions/37346371-support-websocket-connections-on-azure-front-door

**Comment by trgabrie** [comment]
@ShadowXVII any updates on this?

**Comment by joshjohanning** [comment]
We were able to work around this by using the [Azure SignalR service](https://azure.microsoft.com/en-us/services/signalr-service/#solution-architectures) instead of interacting with Front Door's websockets limitation. 

If I recall, the connection is something like this: 
User --> App Service behind Front Door <--> Azure SignalR

**Comment by ShadowXVII** [comment]
> @ShadowXVII any updates on this? 

Was "There is currently no ETA but may come at the end of this year." as at August last year.

Radio silence since.

**Comment by chrish434** [comment]
Just ran into this myself, very much a shame MS hasn't progress this in over 2 years

**Comment by antonmatsiuk** [comment]
It's 2022 already and WSS support after 3 years is still not there, incredible!

**Comment by davidpeden3** [comment]
+1 when will this gaping hole be addressed?

**Comment by jgwinner** [comment]
Agreed, gaping hole. Also, it's not in the decision tree, or not obvious.

**Comment by mweel1** [comment]
Come on Microsoft.  Web Sockets are as standard today as HTTP services.

**Comment by Umeshpaikk** [comment]
Any update on this ask ?

**Comment by darena-patrick** [comment]
Greetings, looking for updates as well

**Comment by sebas3552** [comment]
Another developer here waiting for MS to support Web Sockets on AFD.

**Comment by joshuadmatthews** [comment]
The silence here from Microsoft is concerning. At least tell us you aren't implementing it if you aren't implementing it.

**Comment by darena-patrick** [comment]
Maybe someone at MS Ignite can ask the product team?

**Comment by ee99ee** [comment]
+1 for this

**Comment by dfendit** [comment]
still no redemption ... I will ask the product team and circle back to this thread

**Comment by gball14** [comment]
The team is actively working on enabling to websockets on Azure Front Door. Currently, we are tracking for tentative ETA of March 2023.

**Comment by joeljanke** [comment]
> The team is actively working on enabling to websockets on Azure Front Door. Currently, we are tracking for tentative ETA of March 2023.

Here we are in January 2023.   Still looking like delivery in March 2023 for web sockets @gball14 ?

**Comment by montyplattner** [comment]
@gball14, Following-up on joeljanke's question. Will there be a public preview for this feature in March? Has the roadmap timeline changed at all since your comment in October 2022?

**Comment by mrbergeron83** [comment]
Any update on websocket for front door? I believe this topic is important enough to have a clear channel of communication

**Comment by nagyleventeboldizsar** [comment]
Hello There :) any updates?

**Comment by Ziiimon** [comment]
I would also be very interested to know if there is an update on this.

**Comment by KonstantinKaminski** [comment]
Why is this item Closed?

We are also eagerly waiting on websocket support...

**Comment by guardrex** [comment]
@gball14 ... I'm still maintaining a remark in the official Blazor docs *Threat Mitigation* topic on this. Since the work seems to have been tracked internally, you'll have to tell us (me) here if WebSockets support for Azure Front Door was instituted, possibly just as a public preview, on target last month (March). From the still-open Azure feedback site at ...

https://feedback.azure.com/d365community/idea/c8b1d257-8a26-ec11-b6e6-000d3a4f0789

... it doesn't look like it's been implemented yet ... or at least there was no update from the team there as to what's going on with it. What's the current status?

**Comment by adeturner** [comment]
Any update on the ETA?

**Comment by guardrex** [comment]
I've opened a new GH issue to try and get an update. I recommend to the community to add 👍 (thumbs up) on my opening comment.

### ***There's no need to add comments to the issue, as that just generates lot of annoying email pings for everyone. ONLY add a 👍 (thumbs up) to my opening comment.***

That's all that's needed to add support for the ask to get information on where the feature is on their roadmap.
 
Here's the new issue 👉 https://github.com/MicrosoftDocs/architecture-center/issues/4011

**Comment by mpicard** [comment]
Update?

**Comment by aiakubovich** [comment]
@gball14 

updates???

**Comment by bartvdbraak** [comment]
Why are there literally no updates on this?

**Comment by darena-patrick** [comment]
new issue:
https://github.com/MicrosoftDocs/architecture-center/issues/4011
Also closed with no clear answer

**Comment by ckittel** [comment]
One of the main reasons is that this is an issue tracker for content across all of the Azure Architecture Center and isn't routinely monitored by anyone that deals with product roadmap for AFD.  Occasionally, you'll get someone with some roadmap knowledge that happens to catch it, but it's usually not within any team's remit to monitor this GitHub issue list for product questions.  The original question was a content question, which was totally on topic, but going into timing of feature releases is something the content team here is not positioned to facilitate.

**Comment by calinpavel787** [comment]
Update on my side: I'm able to run the Blazor Server 7.0 on Windows & Linux App Service with Front Door and WAF policy.

**Comment by joshuadmatthews** [comment]
@calinpavel787 through Front Door??

**Comment by darena-patrick** [comment]
I'm able to too- but only if I use Azure SignalR coupled with it

**Comment by joshuadmatthews** [comment]
But where is the Front Door we sockets support? Why are all of these tickets for it closed with seemingly nothing released? Am I missing something? Been watching these for years

**Comment by garrettlondon1** [comment]
Any update here?

**Comment by ckittel** [comment]
I'll say this again, and I'm sorry, but this issue thread is not monitored in an official way that would be used to answer this question. This closed issue in on an issue tracker that is related to reporting bugs in content on <learn.microsoft.com>.  Everyone is welcome to keep asking here for updates on product features, but please don't be expect an answer to this product feature timeline question from a content bug thread.

Matter of fact, if you haven't [seen the announcements](https://learn.microsoft.com/contribute/content/provide-feedback) GitHub issues will be closing down for content teams like ours and this _issue will stop existing_ when that happens to this repo. To that end, I strongly recommend you take this conversation to Microsoft Q&A or another channel to carry on the conversation.

Try one of: <https://learn.microsoft.com/search/?terms=websocket%20front%20door&category=QnA>

> [T]his is an issue tracker for content across all of the Azure Architecture Center and isn't routinely monitored by anyone that deals with product roadmap for AFD. Occasionally, you'll get someone with some roadmap knowledge that happens to catch it, but it's usually not within any product team's remit to monitor this GitHub issue list related to content bugs for product roadmap questions. The original question was a content question, which was totally on topic, but going into timing of feature releases is something the content team here is not positioned to facilitate.

**Comment by ckittel** [comment]
>  I really dread the migration as it looks like the plan is a webform to dump complaints into without any feedback/interaction/community (think "screaming into the void").

I'm sure a lot of folks share this concern both internally and externally (but obviously not all). While I can't speak for anyone else in this universe, you can personally count me as one of them. Dialog is very helpful for me, as a content owner, to be able to get to the root of content issues with the people that take the time to engage with the feedback system and allows everyone to clarify their thoughts and asks through iterative, async dialog.
